students/uss_tomasz/lesson_02_flow_control/main.py

```diff
@@ -0,0 +1,165 @@
+"""
+Lesson2
+https://snakify.org/lessons/if_then_else_conditions/problems/
+"""
+# Problem Rook move
+x_start = int(input("Starting Column of rook movement"))
+y_start = int(input("Starting Row of rook movement"))
+x_finish = int(input("Finishing Column of rook movement"))
+y_finish = int(input("Finishing Row of rook movement"))
+
+if x_start == x_finish or y_start == y_finish:
+    print("YES")
+else:
+    print("NO")
+
+# Problem Chess board - same color
+x_start = int(input("Declare first field: Column"))
+y_start = int(input("Declare first field: Row"))
+x_finish = int(input("Field to check, is it the same color: Column"))
+y_finish = int(input("Field to check, is it the same color: Row"))
+
+if (x_start + y_start) % 2 == (x_finish + y_finish) % 2:
+    print("YES")
+else:
+    print("NO")
+
+# Problem King move
+x_start = int(input("Starting Column of king movement"))
+y_start = int(input("Starting Row of king movement"))
+x_finish = int(input("Finishing Column of king movement"))
+y_finish = int(input("Finishing Row of king movement"))
+
+if abs(x_start - x_finish) < 2 and abs(y_start - y_finish) < 2:
+    print("YES")
+else:
+    print("NO")
+
+# Problem Bishop moves
+x_start = int(input("Starting Column of bishop movement"))
+y_start = int(input("Starting Row of bishop movement"))
+x_finish = int(input("Finishing Column of bishop movement"))
+y_finish = int(input("Finishing Row of bishop movement"))
+
+if abs(x_finish - x_start) == abs(y_finish - y_start):
+    print("YES")
+else:
+    print("NO")
+
+# Problem Queen move
+x_start = int(input("Starting Column of queen movement"))
+y_start = int(input("Starting Row of queen movement"))
+x_finish = int(input("Finishing Column of queen movement"))
+y_finish = int(input("Finishing Row of queen movement"))
+
+if (abs(x_finish - x_start) == abs(y_finish - y_start)) or (x_start == x_finish or y_start == y_finish):
```

Comment:
in python `or` operator has low priority http://www.mathcs.emory.edu/~valerie/courses/fall10/155/resources/op_precedence.html
so some of the brackets above are not needed - this code works in the same way
```
if abs(x_finish - x_start) == abs(y_finish - y_start) or x_start == x_finish or y_start == y_finish:
```

anyway, they can be added for code clarity like in the example here: https://introcs.cs.princeton.edu/python/appendix_precedence/